### PR TITLE
Integration spec n+1 problem

### PR DESCRIPTION
**Pull request by David-Verde**
In this pull request, I've searched for n+1 problems and implemented integration tests.

Task: https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/projects/intergration_spec_and_n_1_project.md

This PR makes the following changes:

Use bullet gem to look for n+1 problems
Write integration tests for the app.
*Type of change
 Bug fix (non-breaking change which fixes an issue)
 Test (change related to testing)
*Checklist
 My code follows the style guidelines of this project (no linters errors)
 I have checked my code and corrected any misspellings
 I have commented on my code
 I have deleted any unused code as comments in my codebase

### CODE REVIEW

File: spec/models/comment_spec.rb [hunk 1]
```diff
@@ -0,0 +1,11 @@
+require 'rails_helper'
+
+RSpec.describe Comment, type: :model do
```
Comment (Munsa1) [inline spec/models/comment_spec.rb:3]: - [ ] Kindly fix the syntax error in `comment_spec.rb` as this is causing a failure of all tests to run.
image

**Review by Munsa1 — CHANGES_REQUESTED** [review]
Hi @David-Verde, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
<FEEL FREE TO CHANGE THIS MESSAGE TO YOUR PERSONAL WORDS>

## Highlights
:heavy_check_mark: Descriptive Readme
:heavy_check_mark: Passing Linters
:heavy_check_mark: Correct Gitflow
## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by David-Verde** [comment]
image
thanks for the suggestion I already fix it.

**Review by mukhammadrizooff — APPROVED** [review]
Hi @David-Verde, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



## Highlights
- No linter errors ✔️ 
- Good implementations ✔️ 
- Descriptive `PR` description and commit history ✔️ 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**